Project | Devise Authorization in Rails

pointerish opened the pull request:
This PR implements the following:

- Models for `User` and `Post`
- Controller for `Post` and `Devise` `Registration`
- Associations between models
- Authentication using the `Devise` gem
- Simple UI prettified with SASS.

I'm aware I don't have enough commits. I worked alone and got so into it that totally forgot to commit bit by bit as it should be done. Apologies for that. I will do my best to not do that again.

teekaytech commented on config/routes.rb:
```diff
@@ -0,0 +1,11 @@
+  get 'users/sign_out'
+  get 'posts/show'
+  devise_for :users
+  resources :posts
```
You should create the application routes using `resources :posts, only: [:new, :create, :index]` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_routes_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))

teekaytech commented on app/controllers/posts_controller.rb:
```diff
@@ -0,0 +1,32 @@
+  end
+
+  def new
+    @post = current_user.posts.build
```
You should initialize an empty post with `@post = Post.new` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_new_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))

Using `@post = current_user.posts.build` will be necessary for the create method, just as you've already done it.

teekaytech commented on app/views/posts/_navbar.html.erb:
```diff
@@ -0,0 +1,26 @@
+    <div class="brand-name">
+        <%= link_to('MembersOnly', root_path) %>  
+    </div>
+    <div class="nav-links">
+      <div class="nav-link">
+        <% if user_signed_in? %> 
+          <%= @post.user.email %>
+        <% else %>
+          <%= link_to('Login', new_user_session_path) %>  
+        <% end %>
+        </div>
+      <div class="nav-link">
+        <% if user_signed_in? %> 
+          <%= link_to('Logout', destroy_user_session_path, method: :delete) %>
+          <% else %>
+            <%= link_to "Create Account", "/users/sign_up" %>
+        <% end %>
+        </div>
+      <div class="nav-link">
+        <% if user_signed_in? %> 
+          <%= link_to('New Post', '/posts/new') %>
+        <% end %>
+      </div>
+    </div>
```
There should be no logic in views. Please use the Rails Helpers to do this.
The articles below might help:
[Where Do I Put My Code?](http://codefol.io/posts/Where-Do-I-Put-My-Code)
[The Beginner’s Guide to Rails Helpers](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers)

teekaytech commented on app/views/posts/index.html.erb:
```diff
@@ -0,0 +1,14 @@
+            <%= link_to(post.title, "posts/show?id=#{post.id}") %>
+          </div>
+          <div class="posted-by">
+            <% if user_signed_in? %> Posted by <%= post.user.email %><% end %>
```
There should be no logic in views. Please use the Rails Helpers to do this.

teekaytech reviewed (changes requested):
## Status: Changes Required :warning: :vertical_traffic_light: 

### Hello team :wave:

__You've done so well in this project.
However, you have to implement some required changes before this project can be approved. Please check my comments below and fix them as appropriate.__

**[ REQUIRED CHANGES ]**

- [x] Update your readme file.
- Your readme is missing almost all the important sections required. Kindly update the readme file to suit our project before submitting another code review request.
- [x] Your project is missing GitHub Actions linters setup.
- You need to set up the appropriate code linter for your project. [This repo](https://github.com/microverseinc/linters-config/tree/master/ror) contains the necessary procedure required to set it up.
- [ ] I understand why you are having a single commit history for your implementation of this project feature, based on your explanation. As you make each of these required changes, always commit often.
- [ ] Make sure that there is no logic in views and that helpers are responsible only for view-related logic. 
- This can be seen in most of your view files, like `posts/_navbar.html.erb`, `posts/index.html.erb`, etc.
- Check the inline comments below for details.
- [ ] I observed that you rendered the `navbar` individually in about 3 to 4 views. You can just render this partial inside the `application.html.erb` so it is not manually repeated in every view. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/sign_in_out_link_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49722686/))


* * *
**Please make the required changes and submit another review request when you are done.**

* * *

**Taofeek Olalere
[Twitter](https://twitter.com/ola_lere) | [Github](https://github.com/teekaytech) | @taofeekolalere on Slack**
__Happy Coding :smiley:__

pointerish commented:
Hello TSEs,

I made the fixes and improvements required:

- Took all logic out of the views and used helpers instead
- Rendering navbar with application.html.erb instead of rendering it individually on my views
- Improved code on `routes.rb` as indicated by the TSE
- Added README.md
- Added linter configuration.

The last two were done right after I submitted the first review but the previous TSE started reviewing it when those were not included just yet. Apologies for that.

javitocor reviewed (changes requested):
# STATUS CHANGES REQUIRED ♻️ 

Hello! 👋 
A great job implementing all the changes requested by the previous TSE 🥇 
However, we still to do something else to get the project approved 💪 
Kindly check my comments below 👇  apply the changes and request a new code review ⏩ 
Good luck! 😺 
## Mandatory changes
- Kindly remove the line '@post = current_user.posts.build if user_signed_in?' in your post controller since it is not necessary
- Commit more and more often

voscarmv reviewed (approved):
# Project status: Approved ✨ 😎👍 💯 ✔️
### You may merge this branch and move forward ⛙🛣

### Questions or comments? Reach me on [Twitter](https://twitter.com/voscarmv) or [LinkedIn](https://linkedin.com/in/oscar-mier) 🧑‍💻🗨️🧑‍💻 
# Happy Hacking! 🧠 ⌨️ 🖱️ 🖥️
